pkg/lib/interaction/nodes/do_create_user.go

```diff
@@ -64,11 +59,37 @@ func (n *NodeDoCreateUser) GetEffects() ([]interaction.Effect, error) {
 			return err
 		}),
 		interaction.EffectOnCommit(func(ctx *interaction.Context, graph *interaction.Graph, nodeIndex int) error {
+			ip := httputil.GetIP(ctx.Request, bool(ctx.TrustProxy))
+			// check the rate limit only and don't take the token yet
+			if !n.BypassRateLimit {
+				pass, _, err := ctx.RateLimiter.CheckToken(interaction.SignupRateLimitBucket(ip))
```

Comment:
Any reason why don't we just call TakeToken?

Reply:
Since the real effects are run in `ctx.Users.AfterCreate`.
1. We want to ensure there is token before running the effects, so we check token here.
2. We only want to take the token if the effects are success. If effects failed (e.g. webhook disallow the operation), we don't take the token, so user can retry immediately. 

If the implementation looks good, let me add more comment in the code to explain.